### ISSUE TITLE
Add streaming ByteBuffer decompress

### ISSUE THREAD

**Issue by DavyLandman** [luben/zstd-jni#27]
It would be nice to extend the ByteBuffer interface with a way to stream through the byte buffer. At the moment, the whole ByteBuffer source is read, and especially with large files, this can require a too large target ByteBuffer. 

The primary use case would be memory mapped byte buffers from nio. These are especially interesting for reading files. I was working on a input stream interface around these and noticed that the interface for ByteBuffers are not streaming aware.

I think it would be nice to have a destination buffer of 128K, and read from the src buffer in a streaming fashion, similar to the input stream.

I'm not sure what would be the best solution, create a version of `decompressDirectByteBuffer ` where the long encodes 2 ints, the bytes read from input, and the bytes written to output? Or is the native zstd streaming interface more complicated than that, and do I need to create a similar class as the `ZstdInputStream`?

**Comment by DavyLandman** [comment]
I've been reading the native code, I now understand why just returning an extra int isn't enough, we have to keep track of the "stream" object instance.

I could see a `ZstdDirectByteBufferInputStream` that offers a InputStream interface over a compressed byte buffer?

or something like this:

```java
class ByteBufferStream {
   public ByteBufferStream(ByteBuffer compressed);
   public boolean hasMore();
   public void read(ByteBuffer uncompressed);
}
```

what do you think, do you see a place for this in zstd-jni? I have no problem turning it into a pull-request, just checking beforehand what direction you would prefer.

**Comment by luben** [comment]
Yes, definitely we should have an ByteBuffer stream interfaces. I haven't done it as the Input/Output Stream interfaces are defined in terms of byte arrays.

**Comment by DavyLandman** [comment]
This has been implemented in #28. So closing the issue.

(@luben, any insights on when we'll get a new release?)

**Comment by luben** [comment]
I release in sync with Yann's zstd. Usually once per month.  I try also to keep on the same version.
